web/src/dialog_widget.ts

```diff
@@ -152,6 +155,32 @@ export function launch(conf: WidgetConfig): void {
 
     const $submit_button = $dialog.find(".dialog_submit_button");
 
+    if (conf.update_submit_on_state_change) {
+        const original_values: {[key: string]: unknown} = {};
+        const current_values: {[key: string]: unknown} = {};
+
+        $submit_button.prop("disabled", true);
+        $(".modal__content :input").each(function () {
+            const property_name = $(this).attr("name") as string;
+
+            original_values[property_name] = $(this).val();
+            current_values[property_name] = $(this).val();
+
+            $(this).on("input", function () {
+                // Check if the new value is different from the current value
+                if ($(this).val() !== current_values[property_name]) {
+                    current_values[property_name] = $(this).val();
+                }
```

Comment:
Could we make this more robust by having `current_values` be read from the DOM each time? 

I worry about missing an event that changes the value but isn't an `input` event (e.g. some live-update) resulting in the logic being wrong. I think the best way to do this would be to just write a function that loops through the inputs and collects their values, and call that function in two places -- when the input is first rendered and again when there's a change to check if it's different.